bin/ci/verify-pr-build.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -u
+
+# Don't attempt to build the following plans. They have resource requirements 
+# or build times that exceed the currently available resources on our CI 
+# infrastructure.
+PLAN_BLACKLIST=(
+ glibc
+ gcc
+ ghc
+ ghc710
+ ghc710-bootstrap
+ ghc80
+ ghc82
+ ghc82-bootstrap
+ ghc84
+ ghc86
+ kubernetes
+ mongodb
+ mysql
+)
+plan="$(basename $1)"
+
+HAB_ORIGIN=ci
+export HAB_ORIGIN
+
+for bl in "${PLAN_BLACKLIST[@]}"; do 
+  if [[ "${plan}" == "${bl}" ]]; then 
+    echo "--- :bangbang: [$plan] Skipping build"
+    # If we're running in buildkite, we also want to annotate the build 
+    if [[ "${CI:-}" == "true" ]]; then 
+      buildkite-agent annotate --style 'warning' ":bangbang: ${plan} found in build blacklist. Skipping build. "
+    fi
+    exit 0;
+  fi
+done
+
+
+echo "--- :key: Generating fake origin key"
+# This is intended to be run in the context of public CI where 
+# we won't have access to any valid signing keys. j
+hab origin key generate "$HAB_ORIGIN"
+
+
+echo "--- :construction: Starting build for $plan"
+# Build with DO_CHECK=true.
+# There are a number of plans that have false-positive tests
+# or "known failing" tests that are currently ignored by maintainers
+# For now this script will be non-enforcing and will always
+# return success. It will be on the maintainers to verify the build
+# and determine if any failures need correction.  In the future
+# this will be turned into an enforcing script, preventing merges
+# if do_check produces any failures.
+
+# We want to ensure that we build from the project root. This 
+# creates a subshell so that the cd will only affect that process
+project_root="$(git rev-parse --show-toplevel)"
+( 
+  cd "$project_root"
+  env DO_CHECK=true hab pkg build "$plan"
```

Comment:
I've had mixed success with running `hab pkg build` from inside a docker container. I think the best thing to do is see if this works, and go from there.

Reply:
10-4.  I know that the chroot studio needs certain privileges to bind mount filesystems, like /proc. As long as the container has the [correct capabilities](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities) for those mounts it should work.  I don't know what the required capabilities are off the top of my head, but if we run into problems we can move this to a different executor.